### ISSUE TITLE
DOC: follow Google's developer documentation style guide 

### ISSUE THREAD

**Issue by martinfleis** [geopandas/geopandas#1896]
We should make sure we use the same documentation style across our docs. @andreaseliasson suggested using [Google developer documentation style guide](https://developers.google.com/style/highlights), which sounds sensible.

We should add it to the contributing docs and revisit the existing user guide and examples to follow this style.

xref #1886

**Comment by rajpratyush** [comment]
@martinfleis can i work on this issue

**Comment by martinfleis** [comment]
@rajpratyush Yes. Start with an expansion of contributing docs if possible so we have a reference. Thanks!

**Comment by shakurgds** [comment]
I am planning to work on this (reading the google documentation style guidelines) and I assume you need more hands. Referring to [xref #1886](https://github.com/geopandas/geopandas/pull/1886), I see the documentation style is not complete. Can I start with the introduction notebook?

**Comment by martinfleis** [comment]
@shakurgds check if there are any open PRs related to this to avoid overlaps. Otherwise, feel free to pick whatever you like! Thanks!

**Comment by Leviob** [comment]
Hi, I'm going to make a pass through the docs and fix a few style issues.
